include/nitro/broken_options/option/group.hpp

```diff
@@ -41,34 +44,189 @@ namespace broken_options
     class group
     {
     public:
-        group(const std::string& name, const std::string& description = std::string(""))
-        : name_(name), description_(description)
+        std::string name;
+        std::string description;
+
+    public:
+        group(std::set<std::string>& all_argument_names, const std::string& name,
+              const std::string& description = std::string(""))
+        : name(name), description(description), all_argument_names_(all_argument_names)
         {
         }
 
-        void add(base& option)
+        group& subgroup(const std::string& name, const std::string& description = std::string(""))
         {
-            options_.emplace(option.name(), std::ref(option));
+            if (std::count(sub_groups_.begin(), sub_groups_.end(), name) == 0)
```

Comment:
This method should be implemented with only one `find()` and zero `count()` calls.